rpcserver.go

```diff
@@ -3312,12 +3314,56 @@ func (r *rpcServer) BurnAsset(ctx context.Context,
 		}
 	}
 
+	// At this point everything completed correctly, so we log this burn in
```

Comment:
Hmm, perhaps we should do this further in the pipeline? So in the same db transaction that we insert the transfer. 

Otherwise, if we crash here, then the burn is never inserted in the db, and we exit in an inconsistent state.

Reply:
that seems to happen within the state machine of the chain porter, deep in the `r.cfg.ChainPorter.RequestShipment` step above

threading through all the params needed for the `InsertBurn` call seems like a bad way to go

Reply:
> threading through all the params needed for the InsertBurn call seems like a bad way to go

Can you think of an alternative? 

We want this visibility to be air tight. If it's possible to just not record a burn all together, then that's problematic. 

Perhaps we need a hybrid migration route where we scan the db to insert the new burn metadata into transfers, then can rely on that information being specified in a transfer? 

Can you explain the issue you see with adding this meta data at the transfers level to it can be inserted when we go to log the parcel?

Reply:
Seems like most of the burn info is available in the Parcel itself, so with the latest push I"m inserting the Burn in the same transaction as the transfer.

Only needed to add the `note string` argument as an extra to some interfaces, but it's minimal